docs/admin/service_manager.md

```diff
@@ -0,0 +1,189 @@
+and Gunicorn the overall performance of the service is increased to allow
+multiple requests to be served at a time.
+
+## Currently supported API endpoints - MasterMind Service Manager
+
+* `POST /v1/stack` - Create a new stack deploying services, creating networks,
+  volumes, secrets and configs defined in the docker-compose file.
+
+* `POST /v1/stack/{name}` - Return the status of an existing stack with the
+  services currently running in the Stack and the total amount of replicas.
+
+* `POST /v1/stack/delete/{name}` - Delete the stack removing services and
+  networks. Note that volumes, secrets and configs are not deleted.
+
+![image alt text](../images/swagger_example.png)
+
+The above image depicts a `POST` request to the `/v1/stack` endpoint. A set of
+parameters are given as strings plus a key-value mapping JSON object for this
+endpoint:
+
+* **name:** Name of the stack - this is equivalent to the stack_name parameter
+  used in a docker stack deploy command;
+
+* **compose-file:** A string containing the entire docker-compose file in yaml
+  format; Variables may be referenced in this file which are obtained from
+  compose-vars; The syntax used for variable substitution is analogous to
+  docker-compose, i.e. `${VARIABLE}`.
+
+* **compose-vars:** Also a string in yaml format containing the a key-value
+  mapping of the variables previously defined.
+
+* **engine-url:** Swarm manager docker engine url. For example:
+  `tcp://swarmmanager.zurich.fiwarelab.org:2376`
+
+* **ca-cert:** A string containing the ca.pem certificate protecting the
+  docker-engine.
+
+* **cert:**  A string containing the cert.pem certificate protecting the
+  docker-engine.
+
+* **cert-key:** A string containing the key.pem certificate protecting the
+  docker-engine.
+
+* **external-files:** An array of JSON objects containing all external files
+  required by the compose-file, .i.e secrets and configs. The JSON object
+syntax should be `{‘filename’: ‘file_content’}`.
+
+The IP address of the Swarm manager can be found in the [Rancher](http://rancher.com/)
+interface under the tab Infrastructure -> Hosts (More details below). Note
+that a swarm may have more than one manager: in this case, any of the swarm
+managers could be chosen. The default port used by Docker engine is 2376 and
+this is the port usually used in the engine-url parameter. *Note that
+responsibility for ensuring this port is open on the VM lies with the user -
+it may require specifically setting up security groups and/or other firewall
+rules to ensure these ports are accessible*.
+
+## Security
+
+In order to safely connect to a remote Docker engine, the Mastermind Service
+Manager requires TLS certificates to protect the connection to the remote
+Docker engine. [This is a common practice in Docker](https://docs.docker.com/engine/security/https/).
+
+[Rancher](http://rancher.com/) makes use of docker-machine which automatically
```

Comment:
From a reader's point of view, rancher seems to come a bit out of the blue in mastermind docs. I don't recall if there's some part giving some background context or relating both. Is there?